From bb21b453413a2f98069fef482761982d1e90234a Mon Sep 17 00:00:00 2001
From: Max Reitz <mreitz@redhat.com>
Date: Mon, 4 Nov 2013 22:32:13 +0100
Subject: [PATCH 20/87] qemu-img create: Emit filename on error

RH-Author: Max Reitz <mreitz@redhat.com>
Message-id: <1383604354-12743-23-git-send-email-mreitz@redhat.com>
Patchwork-id: 55322
O-Subject: [RHEL-7.0 qemu-kvm PATCH 22/43] qemu-img create: Emit filename on error
Bugzilla: 1026524
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Fam Zheng <famz@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

BZ: 1026524

bdrv_img_create generally does not emit the target filename, although
this is pretty important information. Therefore, prepend its error
message with the output filename (if an error occurs).

Signed-off-by: Max Reitz <mreitz@redhat.com>
(cherry picked from commit b70d8c237a0e5e829474c3a12c8783893c4e470e)

Signed-off-by: Max Reitz <mreitz@redhat.com>
---
 qemu-img.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 qemu-img.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/qemu-img.c b/qemu-img.c
index 9fda8cf..0f869f3 100644
--- a/qemu-img.c
+++ b/qemu-img.c
@@ -409,7 +409,7 @@ static int img_create(int argc, char **argv)
     bdrv_img_create(filename, fmt, base_filename, base_fmt,
                     options, img_size, BDRV_O_FLAGS, &local_err, quiet);
     if (error_is_set(&local_err)) {
-        error_report("%s", error_get_pretty(local_err));
+        error_report("%s: %s", filename, error_get_pretty(local_err));
         error_free(local_err);
         return 1;
     }
-- 
1.7.1

